### PR TITLE
[SPARK-13037][ML][PySpark] PySpark ml.recommendation support export/import

### PR DESCRIPTION
**Pull request by vectorijk**
PySpark ml.recommendation support export/import.

### CODE REVIEW

**Comment by vectorijk** [comment]
cc @yanboliang Could you review this one?

File: python/pyspark/ml/recommendation.py [hunk 1]
```diff
@@ -81,6 +82,23 @@ class ALS(JavaEstimator, HasCheckpointInterval, HasMaxIter, HasPredictionCol, Ha
     Row(user=1, item=0, prediction=2.6258413791656494)
     >>> predictions[2]
     Row(user=2, item=0, prediction=-1.5018409490585327)
+    >>> import os, tempfile
+    >>> path = tempfile.mkdtemp()
+    >>> ALS_path = path + "/als"
```
Comment (yanboliang) [inline python/pyspark/ml/recommendation.py]: nit: ```ALS_path``` -> ```als_path```

File: python/pyspark/ml/recommendation.py [hunk 2]
```diff
@@ -81,6 +82,23 @@ class ALS(JavaEstimator, HasCheckpointInterval, HasMaxIter, HasPredictionCol, Ha
...
+    >>> als.save(ALS_path)
+    >>> als2 = ALS.load(ALS_path)
+    >>> als.getMaxIter()
+    5
+    >>> model_path = path + "/als_model"
+    >>> model.save(model_path)
+    >>> model2 = ALSModel.load(model_path)
+    >>> model.rank == model2.rank
+    True
```
Comment (yanboliang) [inline python/pyspark/ml/recommendation.py]: Can we also add test for ```userFactors``` or ```itemFactors```?

Comment (vectorijk) [inline python/pyspark/ml/recommendation.py]: right, I also realized this.

**Comment by yanboliang** [comment]
LGTM

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Comment by vectorijk** [comment]
_off-topic_

Hi Spark Devs!

I was wondering if Spark community would be interested in mentoring students for Google Summer of Code(GSoC) under Apache Software Foundation this year. Last time, I was impressed by the MechCoder's project mentored by @mengxr. Therefore, I look forward to having a chance to do something interesting and continue to contribute to codebase during this summer.

@mengxr, Are you still interested in mentoring a MLlib PySpark related project this summer? If so, I am very willing to brainstorm with you and others on JIRA about what could be probably worked on during this summer. And I might start to write the proposal for GSoC. If there are some pre-GSoC issues related to project, I would love to work on those.

P.S Here is the [post](http://apache-spark-developers-list.1001551.n3.nabble.com/GSoC-Interested-in-GSoC-2016-ideas-td16224.html) I published on dev mailing list.

**Pull request closed without merging**